crates/core/c8y_api/src/json_c8y_deserializer.rs

```diff
@@ -277,14 +277,17 @@ impl C8ySoftwareUpdate {
 impl C8ySoftwareUpdateModule {
     fn get_module_version_and_type(&self) -> (Option<String>, Option<String>) {
         // For C8Y version >= 10.4, refer to the issue #1352
-        if let Some(module_type) = &self.software_type {
-            let version = if self.version.is_empty() || self.version == " " {
-                None
-            } else {
-                Some(self.version.clone())
-            };
-            let module_type = (!module_type.is_empty()).then(|| module_type.clone());
-            return (version, module_type);
+        match &self.software_type {
+            Some(module_type) if !module_type.is_empty() => {
+                let version = if self.version.is_empty() || self.version == " " {
```

Comment:
Wouldn't it be better to trim the whitespace in the version, then check if it is empty?

Reply:
Your suggestion makes the logic smarter. On the other hand, having one space in the version " " comes from our specs.

https://thin-edge.github.io/thin-edge.io/start/software-management/#adding-new-software-into-the-software-repository-of-your-tenant
> thin-edge.io ships a default plugin supporting debian packages from both apt repositories as well as remote locations. If you prefer to use packages from an apt repository, select the Provide a file path option and give an empty space (' ').

So, I think the current implementation makes sense to tell that we address `" "` as special case.

Reply:
I think we need to revisit that spec...a magic single space is very fragile and makes absolutely no sense

Reply:
In some languages (e.g. c#) there is even a string method to check for a null or whitespace string, `String.IsNullOrWhiteSpace()`

Reply:
> Wouldn't it be better to trim leading and trailing whitespace in the version, then check if it is empty?

Indeed better to use `self.version.as_ref().trim().is_empty()` here.

Accepting more than one space simply makes things less error prone for the users, without any contradiction with  the docs telling one space is enough as a workaround for fields that are mandatory on the GUI. Accepting `"NA"` for that purpose might be more explicit.

Reply:
To change the code, I need to make sure what is the expected behaviour. Here I just sumarized what we have now in this PR.

a. software version (mandatory in UI)
* (as of now) If user sets to "latest" or an empty space (' '), both get interpreted to "latest" version to be used. ([reference](https://thin-edge.github.io/thin-edge.io/start/software-management/#adding-new-software-into-the-software-repository-of-your-tenant)) 

b. software type (mandatory in UI since 10.14)
* (new in this PR) If software type is not given or having no value (''), falling back to `::` logic. No special care for an empty space.

c. software URL (mandatory in UI)
* (as of now) An empty space (' ') gets interpreted to use the device's default repository.

From the discussion so far, we will trim empty spaces (so it will not matter if it's ONE space or more). But, should we still mention about the empty space stuff in our user guide?

Reply:
> From the discussion so far, we will trim empty spaces (so it will not matter if it's ONE space or more).

Perfect

> But, should we still mention about the empty space stuff in our user guide?

Yes, we should. This is notably useful when updating a package using the package repository configured on the device.

Reply:
Applied in 4188f534f8a8dad27c5413bab850b0f3d326615a